src/Tools/Source/RunTests/ProcessTestExecutor.cs

```diff
@@ -91,7 +101,10 @@ private string GetResultsFilePath(AssemblyInfo assemblyInfo)
                     lowPriority: false,
                     displayWindow: false,
                     captureOutput: true,
-                    cancellationToken: cancellationToken);
+                    cancellationToken: cancellationToken,
+                    environmentVariables: environmentVariables,
+                    onProcessStartHandler: (process) => ProcDumpRunner.StartProcDump(process.Id, process.ProcessName, outputDirectory, Logger.Log)
```

Comment:
We use -w option in procdump. This means we wait for the process to start. If we could not find the process, I think the instance of the procdump will be executing during the whole test session. Will this make a trouble?

---
In reply to: [122364602](https://github.com/dotnet/roslyn/pull/20129#discussion_r122364602) [](ancestors = 122364602)